Fix aliased default options

tbekolay opened the pull request:
I recently updated to interact.js 1.3.0 so I could use `restrictSize` and `restrictEdges` when resizing. However, when I updated, I got the following behavior when resizing an element that was working correctly with version 1.2.9.

![interact](https://user-images.githubusercontent.com/203709/34012471-e0efd8aa-e0e1-11e7-98ac-67f119d22231.gif)

Resizing from the top edge is completely messed up. I tried to reproduce this in isolation in [this codepen](https://codepen.io/tbekolay/pen/MrWQGJ), but in isolation this worked fine.

I dug a little deeper, and as it turned out, the top edge was being snapped down to nothing because of what was set in `restrictEdges.outer`. I found this odd because I had not set `restrictedEdges.outer` for this svg rect. However, I _had_ set `restrictEdges.outer` in another file. As it turns out, I had set `restrictEdges.outer` to an element that was on the bottom of the screen, and since this setting bled out to the svg rect, resizing from the top immediately tried to respect the restriction, which was at the very bottom of the page.

This was an annoying bug to track down, but fortunately it was relatively easy to figure out the fix. In `Interactable.js`, `extend` is used in several places to copy defaults. However, `extend` does not make a copy of what it is extending, which in [this specific case](https://github.com/taye/interact.js/blob/master/src/Interactable.js#L63) was causing my problem. The per action default object was being set (aliased) and then modified by the first interactable.

While the bug was somewhat easy to fix, it was unfortunately not very easy for me to test. I could not find another test that made an `Interactable`, so I did my best to make one. It took quite a few iterations to get something that would run and reproduce the error. I will defer to you, @taye, to fix up the test if it could be made better. But it would be great to have the bug fixed so I can use `restrictEdges` and `restrictSize` in my project :)

taye commented:
Thank you, @tbekolay! I spotted this error while I was refactoring some things and have already fixed it on the master branch.

I've created a "v1.3-bugfix" branch so if you change the merge target from "master" to "v1.3-bugfix", then I can accept this pull request and release `v1.3.3`. I'll also try to merge into master manually to use your tests and clone.js module.

tbekolay commented:
That all sounds great, thanks! I changed the target of this PR and rebased my branch to `v1.3-bugfix`.

I had to make a small change to my test, but it passes now; unfortunately I get a test failure in `Interaction.start` now that I can't figure out... here's the test log:

```
  Interactable copies and extends defaults

    ✔ specified options are properly set
    ✔ default options are properly set
    ✔ defaults are not aliased
    ✔ modifying defaults does not affect constructed interactables

  Interactable copies and extends per action defaults

    ✔ specified options are properly set
    ✔ default options are properly set
    ✔ defaults are not aliased
    ✔ modifying defaults does not affect constructed interactables

  Interaction constructor

    ✔ interaction.prepared is an object
    ✔ interaction.downPointer is an object
    ✔ interaction.prevCoords set to zero
    ✔ interaction.curCoords set to zero
    ✔ interaction.startCoords set to zero
    ✔ interaction.pointerDelta set to zero
    ✔ interaction.pointerType is set
    ✔ interaction.pointers is an array
    ✔ interaction.pointers is empty
    ✔ interaction.pointerIds is an array
    ✔ interaction.pointerIds is empty
    ✔ interaction.downTargets is an array
    ✔ interaction.downTargets is empty
    ✔ interaction.downTimes is an array
    ✔ interaction.downTimes is empty
    ✔ interaction.pointerIsDown is false
    ✔ interaction.pointerWasMoved is false
    ✔ interaction._interacting is false
    ✔ interaction.mouse is false
    ✔ new Interaction is pushed to scope.interactions

  Interaction.getPointerIndex

    ✔ should be equal
    ✔ should be equal
    ✔ should be equal
    ✔ should be equal
    ✔ should be equal

  Interaction.updatePointer


  no existing pointers

    ✔ interaction.pointers == [pointer]
    ✔ interaction.pointerIds == [pointer.pointerId]
    ✔ new pointer is at index 0

  new pointer with exisiting pointer

    ✔ interaction.pointers == [pointer]
    ✔ interaction.pointerIds == [pointer.pointerId]
    ✔ new pointer index is n - 1

  update existing pointers

    ✔ number of pointers is unchanged
    ✔ new pointer object !== old pointer object
    ✔ pointerIds are identical
    ✔ new pointer object !== old pointer object
    ✔ pointerIds are identical
    ✔ new pointer object !== old pointer object
    ✔ pointerIds are identical

  Interaction.removePointer

    ✔ first of 4 - remaining interaction.pointers is correct
    ✔ first of 4 - remaining interaction.pointerIds is correct
    ✔ first of 4 - remaining interaction.downTargets is correct
    ✔ first of 4 - remaining interaction.downTimes is correct
    ✔ middle of 3 - remaining interaction.pointers is correct
    ✔ middle of 3 - remaining interaction.pointerIds is correct
    ✔ middle of 3 - remaining interaction.downTargets is correct
    ✔ middle of 3 - remaining interaction.downTimes is correct
    ✔ last of 2 - remaining interaction.pointers is correct
    ✔ last of 2 - remaining interaction.pointerIds is correct
    ✔ last of 2 - remaining interaction.downTargets is correct
    ✔ last of 2 - remaining interaction.downTimes is correct
    ✔ final - remaining interaction.pointers is correct
    ✔ final - remaining interaction.pointerIds is correct
    ✔ final - remaining interaction.downTargets is correct
    ✔ final - remaining interaction.downTimes is correct

  Interaction.pointerDown

    ✔ downEvent is not updated
    ✔ pointer is added
    ✔ downTargets is not updated
    ✔ downTimes is not updated
    ✔ downPointer is not updated
    ✔ startCoords are not modified
    ✔ curCoords are not modified
    ✔ prevCoords are not modified
    ✔ pointerIsDown
    ✔ !pointerWasMoved
    ✔ pointer in down signal arg
    ✔ event in down signal arg
    ✔ eventTarget in down signal arg
    ✔ pointerIndex in down signal arg
    ✔ downEvent is updated
    ✔ downTargets is updated
    ✔ downTimes is updated
    ✔ startCoords are set to pointer
    ✔ curCoords are set to pointer
    ✔ prevCoords are set to pointer
    ✔ down signal was fired again
    ✔ pointerIsDown
    ✔ pointerWasMoved should always change to false

  Interaction.start

    ✔ do nothing if !pointerIsDown
    ✔ do nothing if too few pointers are down
    ✔ do nothing if already interacting
/home/tbekolay/Code/interact.js/src/utils/getOriginXY.js:9
__cov_5nhc$g7Cnh5FiGLHBi_8yA.s['1']++;const {resolveRectLike,rectToXY}=require('./rect');__cov_5nhc$g7Cnh5FiGLHBi_8yA.s['2']++;module.exports=function(target,element,action){__cov_5nhc$g7Cnh5FiGLHBi_8yA.f['1']++;__cov_5nhc$g7Cnh5FiGLHBi_8yA.s['3']++;const actionOptions=target.options[action];__cov_5nhc$g7Cnh5FiGLHBi_8yA.s['4']++;const actionOrigin=(__cov_5nhc$g7Cnh5FiGLHBi_8yA.b['1'][0]++,actionOptions)&&(__cov_5nhc$g7Cnh5FiGLHBi_8yA.b['1'][1]++,actionOptions.origin);__cov_5nhc$g7Cnh5FiGLHBi_8yA.s['5']++;const origin=(__cov_5nhc$g7Cnh5FiGLHBi_8yA.b['2'][0]++,actionOrigin)||(__cov_5nhc$g7Cnh5FiGLHBi_8yA.b['2'][1]++,target.options.origin);__cov_5nhc$g7Cnh5FiGLHBi_8yA.s['6']++;const originRect=resolveRectLike(origin,target,element,[(__cov_5nhc$g7Cnh5FiGLHBi_8yA.b['3'][0]++,target)&&(__cov_5nhc$g7Cnh5FiGLHBi_8yA.b['3'][1]++,element)]);__cov_5nhc$g7Cnh5FiGLHBi_8yA.s['7']++;return(__cov_5nhc$g7Cnh5FiGLHBi_8yA.b['4'][0]++,rectToXY(originRect))||(__cov_5nhc

TypeError: Cannot read property 'TEST' of undefined
    at module.exports (/home/tbekolay/Code/interact.js/src/utils/getOriginXY.js:9:285)
    at new InteractEvent (/home/tbekolay/Code/interact.js/src/InteractEvent.js:9:815)
    at firePrepared (/home/tbekolay/Code/interact.js/src/actions/base.js:9:1450)
    at /home/tbekolay/Code/interact.js/src/actions/base.js:9:509
    at Signals.fire (/home/tbekolay/Code/interact.js/src/utils/Signals.js:9:1536)
    at Interaction.start (/home/tbekolay/Code/interact.js/src/Interaction.js:9:3900)
    at Test.t (/home/tbekolay/Code/interact.js/tests/Interaction.js:283:15)
    at Test.bound [as _cb] (/home/tbekolay/Code/interact.js/node_modules/tape/lib/test.js:64:32)
    at Test.run (/home/tbekolay/Code/interact.js/node_modules/tape/lib/test.js:83:10)
    at Test.bound [as run] (/home/tbekolay/Code/interact.js/node_modules/tape/lib/test.js:64:32)
```
